### PR TITLE
[DCJ-56] Add DAA delete endpoint

### PR DESCRIPTION
**Pull request by aarohinadkarni**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-56

### Summary
Adds an endpoint to delete a DAA and corresponding DAA-LC relationships & DAA-DAC relationships with appropriate tests.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java [hunk 1]
```diff
@@ -202,6 +202,26 @@ public Response deleteDaaForUser(
     }
   }
 
+  @DELETE
+  @Path("{daaId}")
+  @Produces("application/json")
+  @RolesAllowed({ADMIN})
+  public Response deleteDaa(
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java]: While this covers the admin case, I can imagine the need for a chairperson version of this that isn't required at the moment. For that reason, we should probably name this something more appropriate like `adminDeleteDaa`.

File: src/main/java/org/broadinstitute/consent/http/db/DaaDAO.java [hunk 2]
```diff
@@ -191,4 +191,11 @@ INSERT INTO dac_daa (dac_id, daa_id)
       """)
   List<Integer> findDaaDatasetIdsByUserId(@Bind("userId") Integer userId);
 
+  @SqlUpdate("""
+    WITH lc_deletes AS (DELETE FROM lc_daa lc_daa WHERE lc_daa.daa_id = :daaId),
+    dac_delete AS (DELETE FROM dac_daa dac_daa WHERE dac_daa.daa_id = :daaId)
+    DELETE FROM data_access_agreement WHERE daa_id = :daaId
+    """)
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/db/DaaDAO.java]:
```suggestion
  @SqlUpdate("""
    WITH lc_deletes AS (DELETE FROM lc_daa WHERE lc_daa.daa_id = :daaId),
    dac_delete AS (DELETE FROM dac_daa WHERE dac_daa.daa_id = :daaId)
    DELETE FROM data_access_agreement WHERE daa_id = :daaId
    """)
```

File: src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java [hunk 3]
```diff
@@ -202,6 +202,26 @@ public Response deleteDaaForUser(
...
+      @Auth AuthUser authUser,
+      @PathParam("daaId") Integer daaId) {
+    try {
+      daaService.findById(daaId);
+      User user = userService.findUserByEmail(authUser.getEmail());
+      if (!user.hasUserRole(UserRoles.ADMIN)) {
+        return Response.status(Status.FORBIDDEN).build();
+      }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java]: We shouldn't need this check - this is handled with `@RolesAllowed({ADMIN})`

**Review by fboulnois — APPROVED** [review]
Overall looks reasonable, one recommendation:

File: src/main/resources/assets/paths/daaByDaaId.yaml [hunk 4]
```diff
@@ -22,3 +22,24 @@ get:
...
+  description: Deletes Data Access Agreement by Id
+  parameters:
+    - name: daaId
+      in: path
+      description: The DAA ID
+      required: true
+      schema:
+        type: integer
+  tags:
+    - DAA
```
Comment (rushtong) [inline src/main/resources/assets/paths/daaByDaaId.yaml:36]: This could also go in the `Admin` section as well

**Review by rushtong — COMMENTED** [review]
Looks good, see comments inline 👍🏽

**Review by rushtong — APPROVED** [review]
Thank you for the changes 👍🏽

**Pull request merged**